cmd/snap-confine/snap-confine.c

```diff
@@ -231,32 +232,35 @@ int main(int argc, char **argv)
 			      snap_instance);
 			struct sc_mount_ns *group = NULL;
 			group = sc_open_mount_ns(snap_instance, 0);
-			if (sc_create_or_join_mount_ns(group, &apparmor,
-						       base_snap_name,
-						       snap_instance) ==
-			    EAGAIN) {
-				// If the namespace was stale and was discarded we just need to
-				// try again. Since this is done with the per-snap lock held
-				// there are no races here.
-				if (sc_create_or_join_mount_ns(group, &apparmor,
-							       base_snap_name,
-							       snap_instance) ==
-				    EAGAIN) {
-					die("unexpectedly the namespace needs to be discarded again");
+			int retval = sc_join_preserved_ns(group, &apparmor,
+							  base_snap_name,
+							  snap_instance);
+			if (retval == ESRCH) {
+				/* Stale mount namespace discarded or no mount namespace to
```

Comment:
Please use `// ..` comment style to be consistent with the rest of the code.

Reply:
So maybe we could hold off this change and apply it everywhere along with clang format? 

If you really want to use `/*` then it should probably be:
```
/*
 * ...
 */
```
Although, I see more `// ..` multiline comments than `/* ..  */`

Reply:
I can look at clang-format (I'd love to move away from indent) but I feat that will just be a bit painful on history (I would be OK but I don't think people will let me land that change). For now I think we should just ... write comments :)